sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryPartA.cs

```diff
@@ -70,7 +70,18 @@ internal static string GetOperationName(Activity activity, ref AzMonList partBTa
             var httpMethod = AzMonList.GetTagValue(ref partBTags, SemanticConventions.AttributeHttpMethod)?.ToString();
             if (!string.IsNullOrWhiteSpace(httpMethod))
             {
-                return $"{httpMethod} {activity.DisplayName}";
+                var httpRoute = AzMonList.GetTagValue(ref partBTags, SemanticConventions.AttributeHttpRoute)?.ToString();
+                // ASP.NET instrumentation assigns route as {controller}/{action}/{id} which would result in the same name for different operations.
+                // To work around that we will use path from httpUrl.
+                if (!string.IsNullOrEmpty(httpRoute) && !httpRoute.StartsWith("{controller}", StringComparison.OrdinalIgnoreCase))
```

Comment:
Most enterprise application uses `Area` in MVC, where the routing pattern will look like `{area:exists}/{controller=Home}/{action=Index}/{id?}`
Ref: https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/areas?view=aspnetcore-5.0#areas-for-controllers-with-views

Are we worried about this scenario?

Reply:
This question should not block the PR. We could discuss this separately.